### PR TITLE
Angular checks 🍫

### PR DESCRIPTION
**Pull request by moralesbang**
## Summary

Added checks for Angular projects.

**Notes:**
- Folder structure verified here is take from https://github.com/Wolox/tech-guides/pull/211.
- For check if the project use jest the test script in package.json should looks like:
```
{
  scripts: {
    test: 'jest'
  }
}
```

As fallback the check is looking for `import 'jest-preset-angular' in src/test.ts`. Currently the existence of Karma or Jasmine isn't checked.

- Please review if limits values are correct for angular checks.
- For check if project us `ngRx` to manage global information, I'm just checking for `@ngrx/store` dependency. This can be improved further.

## Screenshots

[Change!] Show the screenshots of the views you modified.

## Trello Card

https://trello.com/c/TTQ4rXPq

### CODE REVIEW

File: src/dependenciesChecks/index.js [hunk 1]
```diff
@@ -40,17 +40,25 @@ module.exports = async (installInfo, testPath) => {
...
+      value: packages.some(({ moduleName }) => moduleName === '@ngrx/store')
+    });
+  } else {
+    dependenciesResults.push({
+      metric: dependenciesMetrics.AXIOS_APISAUCE,
+      description: 'Esta instalado axios o apisauce en el proyecto',
+      value: packages.some(({ moduleName }) => moduleName === 'axios' || moduleName === 'apisauce')
+    });
+  }
+
```
Comment (LucasZibell) [inline src/dependenciesChecks/index.js:62]: The if is unnecesary. You can always push the result, and then in each summary we look for what we need.
In react not having `@ngrx/store` will not affect

File: src/envChecks/index.js [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
 const envMetrics = require('./constants');
 
-module.exports = async testPath => {
+module.exports = async (testPath, techChecks) => {
   const envresult = [];
-  const results = await findSync('process.env.', testPath, '.js$');
+  let exitsEnvFile = false;
+
+  if (techChecks === 'angular') {
+    exitsEnvFile = fs.existsSync(`${testPath}/src/environments/environment.ts`);
```
Comment (LucasZibell) [inline src/envChecks/index.js:13]: set the await here

File: src/envChecks/index.js [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
+  } else {
+    const results = await findSync('process.env.', testPath, '.js$');
+    exitsEnvFile = !!analyzeMatches(results).length;
+  }
+
   envresult.push({
     metric: envMetrics.ENV_IS_USED,
     description: 'Se utiliza un .env en el proyecto',
-    value: !!analyzeMatches(results).length
+    value: await exitsEnvFile
```
Comment (LucasZibell) [inline src/envChecks/index.js:22]: Remove this await

File: src/frontEnd/angularChecks/index.js [hunk 4]
```diff
@@ -1,4 +1,90 @@
...
+        ngForUseTrackBy = Object.values(result).every(({ line }) =>
+          ['trackBy', 'ngForTrackBy'].includes(line)
+        );
+      });
+
+      return ngForUseTrackBy;
+    })()
+  });
+
+  console.log(angularResult);
```
Comment (LucasZibell) [inline src/frontEnd/angularChecks/index.js:87]: Remove console.log

File: src/frontEnd/angularChecks/index.js [hunk 5]
```diff
@@ -1,4 +1,90 @@
...
+  });
+  angularResult.push({
+    metric: angularMetrics.USE_HTTP_CLIENT,
+    description: 'Usa http-client para la configuración del api.service.ts',
+    value: apiConfigFile && apiConfigFile.includes(HTTP_CLIENT_IMPORT)
+  });
+  angularResult.push({
+    metric: angularMetrics.SERVICE_PER_SCREEN,
+    description: 'Cada screen tiene un service.ts',
+    value: screensFolder && everyScreenHasService(screensFolder)
```
Comment (LucasZibell) [inline src/frontEnd/angularChecks/index.js:54]: `everyScreenHasService` does not recieve a param

File: src/frontEnd/angularChecks/index.js [hunk 6]
```diff
@@ -1,4 +1,90 @@
...
+  const componentFilePaths = Object.keys(clocReport).filter(filepath => /.component.ts$/.test(filepath));
+  const templateFilePaths = Object.keys(clocReport).filter(filepath => /.component.html$/.test(filepath));
+
+  const everyScreenHasService = () => {
+    const screens = screensFolder.reduce(
+      (result, folder) => Object.assign(result, { [folder]: fs.readdirSync(path.join(screensPath, folder)) }),
+      {}
+    );
+    return Object.values(screens).every(folderHasService);
+  };
```
Comment (LucasZibell) [inline src/frontEnd/angularChecks/index.js:32]: You don't need and Object here.. create an array with all the folders
```
const everyScreenHasService = screensFolder.reduce(...).every(folderHasService)
```

File: src/frontEnd/angularChecks/index.js [hunk 7]
```diff
@@ -1,4 +1,90 @@
...
+      let ngForUseTrackBy = false;
+
+      findSync('ngFor', path.join(testPath, 'src/app'), 'component.html$').then(result => {
+        ngForUseTrackBy = Object.values(result).every(({ line }) =>
+          ['trackBy', 'ngForTrackBy'].includes(line)
+        );
+      });
+
+      return ngForUseTrackBy;
+    })()
```
Comment (LucasZibell) [inline src/frontEnd/angularChecks/index.js:84]: Why are you creating a function just to call it here?

File: src/frontEnd/angularChecks/index.js [hunk 8]
```diff
@@ -1,4 +1,90 @@
...
+  });
+
+  console.log(angularResult);
+
   return angularResult;
```
Comment (LucasZibell) [inline src/frontEnd/angularChecks/index.js:89]: Please add blank spaces between every `angularResult.push `

File: src/linterChecks/index.js [hunk 9]
```diff
@@ -3,6 +3,11 @@ const { getLinterErrorCount } = require('./utils');
 
 module.exports = (testPath, tech) => {
   const eslintResponse = [];
+
+  if (tech === 'angular') {
+    return eslintResponse;
+  }
```
Comment (LucasZibell) [inline src/linterChecks/index.js:9]:
```suggestion
  if (tech === 'angular') return eslintResponse;
```

Comment (moralesbang) [reply src/linterChecks/index.js:9]: Current `eslint-config-wolox` enforce to use that way.

File: src/frontEnd/angularChecks/index.js (hunk 7 above)
Comment (moralesbang) [reply src/frontEnd/angularChecks/index.js:84]: I decided move this to a util function.

Comment (moralesbang) [reply src/frontEnd/angularChecks/index.js:94]: For keep it simple 😄

**Pull request merged**